samples/matter/thermostat/src/temperature_measurement/real_temp.cpp

```diff
@@ -23,15 +31,39 @@ constexpr chip::EndpointId mTemperatureMeasurementEndpointId = 1;
 
 RealSensor::RealSensor()
 {
-	BindingHandler::GetInstance().Init();
+	BindingHandler::Init();
 }
 
 void RealSensor::TemperatureMeasurement()
 {
 	BindingHandler::BindingData *data = Platform::New<BindingHandler::BindingData>();
-	data->clusterId = Clusters::TemperatureMeasurement::Id;
-	data->localEndpointId = mTemperatureMeasurementEndpointId;
-	data->isGroup = BindingHandler::GetInstance().IsGroupBound();
-	DeviceLayer::PlatformMgr().ScheduleWork(BindingHandler::ThermostatWorkerHandler,
-						reinterpret_cast<intptr_t>(data));
+	data->ClusterId = Clusters::TemperatureMeasurement::Id;
+	data->EndpointId = mTemperatureMeasurementEndpointId;
+	data->IsGroup = BindingHandler::IsGroupBound();
+	data->InvokeCommandFunc = TemperatureMeasurementReadHandler;
+	BindingHandler::RunBoundClusterAction(data);
+}
+
+void RealSensor::TemperatureMeasurementReadHandler(const EmberBindingTableEntry &binding,
+						   OperationalDeviceProxy *deviceProxy,
+						   BindingHandler::BindingData &bindingData)
+{
+	auto onSuccess = [dataPointer = Platform::New<BindingHandler::BindingData>(bindingData)](
+				 const ConcreteDataAttributePath &attributePath, const auto &dataResponse) {
+		ChipLogProgress(NotSpecified, "Read Temperature Sensor attribute succeeded");
+		int16_t responseValue = dataResponse.Value();
```

Comment:
Are you sure you don't need to check if `dataResponse.HasData()`? The attribute is nullable.